src/context/user_context.js

```diff
@@ -0,0 +1,114 @@
+import React, { useContext, useEffect, useState } from 'react';
+import axios, { AxiosError } from 'axios';
+
+// Since this rootURL is used throughout the app, it might make sense to instead have a file for "Constants" that can be imported wherever needed
+const rootUrl = 'https://ecommerce-6kwa.onrender.com';
+
+const UserContext = React.createContext();
```

Comment:
This creates a context object. We could also pass it an initial value (eg: `React.createContext({ handleLogin: () => {}, handleRegister: () => {}, handleLogout: () => {}, currentUser : null, authLoading: false, authError: false })`

It returns an object that has a `Provider` property for setting context values.